### PR TITLE
[FR-148] NextJS Forums 

### PR DESCRIPTION
**Pull request by tylergeorges**

### CODE REVIEW

File: src/hooks/messagesQuery/index.ts [hunk 1]
```diff
@@ -383,6 +385,40 @@ export const threadsQuery = graphql(`
...
+          appliedTags
+          canSend
+
+          firstMessage: messageBunch(onlyFetchFirstMessage: true) {
+            messages {
+              ...BaseMessage
+            }
+          }
+
+          messageBunch(limit: 1) {
```
Comment (mason-rogers) [inline src/hooks/messagesQuery/index.ts:407]: I assume the two messageBunch queries are to fetch the first & also most recent message?

File: src/util/short.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// based on https://github.com/researchgate/moment-shortformat/blob/master/moment-shortformat.js
+// Copyright (c) 2015 ResearchGate GmbH, MIT license
+
+import moment, { Moment } from 'moment';
+
+const minute = 6e4;
```
Comment (mason-rogers) [inline src/util/short.ts:6]: What unit is this value supposed to be? I'd prefer if these were defined in a more readable way

e.g `const minute = 60 * 1000`

**Review by mason-rogers — CHANGES_REQUESTED** [review]

File: src/hooks/messagesQuery/index.ts (hunk 1 above)
Comment (advaith1) [reply src/hooks/messagesQuery/index.ts:407]: yeah this is how the current implementation works (though backend ignores the limit param)

**Review by mason-rogers — COMMENTED** [review]
Looks good! Just one question

**Review by mason-rogers — APPROVED** [review]

**Review by advaith1 — CHANGES_REQUESTED** [review]
opening forum threads seems to not work

image

File: src/components/Core/TextChannelContainer/styles.ts [hunk 3]
```diff
@@ -170,7 +168,7 @@ export const TextBoxWrapper = styled.withConfig({
   flexDirection: 'column',
 
   width: 'calc(100% - 32px)',
-  maxHeight: '50vh',
+  // maxHeight: '50vh',
```
Comment (advaith1) [inline src/components/Core/TextChannelContainer/styles.ts:171]: commented out code

**Review by advaith1 — CHANGES_REQUESTED** [review]

File: src/components/Header/ForumHeader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  const currentChannel = guildChannels[channelId];
+
+  if (!currentChannel) return null;
+
+  return (
+    <Header>
+      <Hamburger />
+
+      <Styles.ChannelNameIconWrapper>
+        <Icons icon="FourmChannelIcon" color="channel" marginX="xs" size="sm" />
```
Comment (advaith1) [inline src/components/Header/ForumHeader/index.tsx:22]: typo (also needs to be fixed elsewhere

**Review by advaith1 — CHANGES_REQUESTED** [review]

**Review by advaith1 — APPROVED** [review]

**Pull request merged**